custom_components/better_thermostat/climate.py

```diff
@@ -760,11 +769,11 @@ async def _async_trv_changed(self, event):
 					# if new setpoint is lower than the min setpoint, set it to the min setpoint
 					_overwrite_thermostat_update = False
 					if _new_heating_setpoint < self._min_temp:
-						_new_heating_setpoint = self.min_temp
+						_new_heating_setpoint = self._min_temp
 						_overwrite_thermostat_update = True
 					# if new setpoint is higher than the max setpoint, set it to the max setpoint
-					if _new_heating_setpoint > self.max_temp:
-						_new_heating_setpoint = self.max_temp
+					if _new_heating_setpoint > self._max_temp:
```

Comment:
Shoudn't we read via the function? Otherwise we might set None here. The function do test that and otherwise imports from super().

If you want to use it like that, you have to check for none in self._max_temp. Since int and None, cannot be compared with >, <, etc.